src/services/arweave/WarpDataProvider.ts

```diff
@@ -79,4 +96,97 @@ export class WarpDataProvider implements SmartweaveDataProvider {
     const state = await this.getContractState(id);
     return state?.balances[wallet.toString()] ?? 0;
   }
+  /**
+   *
+   * @param srcCodeTransactionId - ArweaveTransactionId of the source code used to deploy contract
+   * @param initialState - typeof ANTContractState or ArweaveTransactionID - depending on the stateType chosen, either the json initial state of the contract to deploy, or the on-chain states TXID
+   * @param stateType - type of 'TXID' | 'TAG' | 'DATA' - TXID indicates use of a prexisting state on-chain, TAG and DATA use json states locally. DATA puts the state in the data key of the transaction, TAG puts the state in the tag.
+   * @returns
+   */
+  async deployContract({
+    srcCodeTransactionId,
+    initialState,
+    stateType,
+    tags = [],
+  }: {
+    srcCodeTransactionId: ArweaveTransactionID;
+    initialState: ANTContractState | ArweaveTransactionID;
+    stateType: 'TXID' | 'TAG' | 'DATA';
+    tags: TransactionTag[];
+  }): Promise<string | undefined> {
+    const stateSize = new Blob([JSON.stringify(initialState)]).size;
+    const tagSize = new Blob([JSON.stringify(tags)]).size;
```

Comment:
might be helpful as a `util` function

Reply:
also, why are we casting it to an Array? we could also use buffers, but would likely need a polyfill for the browser

Reply:
we are already polyfilling buffer due to other deps so we can if its better

Reply:
`Buffer.byteLength(JSON.stringify(initialState)`; or `Buffer.from(JSON.stringify(initalState).length`